1.16.5/src/main/java/io/github/axolotlclient/modules/hypixel/bedwars/upgrades/TrapUpgrade.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright © 2021-2023 moehreag <moehreag@gmail.com> & Contributors
+ *
+ * This file is part of AxolotlClient.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * For more information, see the LICENSE file.
+ */
+
+package io.github.axolotlclient.modules.hypixel.bedwars.upgrades;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import io.github.axolotlclient.AxolotlClientConfig.Color;
+import io.github.axolotlclient.modules.hypixel.bedwars.BedwarsMode;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+/**
+ * @author DarkKronicle
+ */
+
+public class TrapUpgrade extends TeamUpgrade {
+
+	private final static Pattern[] REGEX = {
+		Pattern.compile("^\\b[A-Za-z0-9_§]{3,16}\\b purchased (.+) Trap\\s*$"),
```

Comment:
Remove redundant ones and I also don't think `It's a trap!` gets triggered. But I can never remember how the regex should go for that one.

Reply:
I'm not entirely sure either..